.github/workflows/build-vscode-extension.yml

```diff
@@ -35,20 +35,23 @@ jobs:
         shell: pwsh
         env:
           GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      - run: scripts/update-vscode-releases.ps1 -version "${{ steps.last_release.outputs.RELEASE_VERSION }}" -filePath "./vscode/microsoft-kiota/package.json" -online
+      - run: scripts/update-vscode-releases.ps1 -version "${{ steps.last_release.outputs.RELEASE_VERSION }}" -packageJsonFilePath "./vscode/microsoft-kiota/package.json" -runtimeFilePath "./vscode/microsoft-kiota/src/kiotaInterop/runtime.json" -online
         shell: pwsh
       - name: Install dependencies
         run: npm ci
         working-directory: vscode/microsoft-kiota
       - run: npm run lint
         working-directory: vscode/microsoft-kiota
+      - name: run unit tests
+        run: npm run test:unit
+        working-directory: vscode/microsoft-kiota
       - name: run tests(linux)
         if: runner.os == 'Linux'
-        run: xvfb-run -a npm run test-with-coverage
+        run: xvfb-run -a npm run test:coverage
```

Comment:
leaving a note so we can better understand what's the situation with coverage.

Reply:
I got to go through it. The changes introduced by this PR: https://github.com/microsoft/kiota/pull/6171 removed the filter that keeps the `vscode` folder outside of the dotnet sonarqube checks. Without this filter the vscode extension will definitely fail the checks because the workflow uses dotnet to test.

Some more configuration work will be required to bring in the report from the sonarqube checks that are already placed inside the `build-vscode-extension.yml` similar to the comment that comes in when the dotnet side is checked. Currently what we have is the production of the code coverage and a connection to the sonarqube host and not the handover back to the PR.

This work is however outside the scope of this PR and may need its own item and prioritisation